logstash-core/lib/logstash/agent.rb

```diff
@@ -274,4 +298,13 @@ def clean_state?
   def setting(key)
     @settings.get(key)
   end
+
+  def init_pipeline_reload_metrics(id)
+    namespace = @reload_metric.namespace([id.to_sym, :reloads])
+    namespace.increment(:successes, 0)
+    namespace.increment(:failures, 0)
+    namespace.gauge(:last_error, nil)
+    namespace.gauge(:last_success_timestamp, nil)
+    namespace.gauge(:last_failure_timestamp, nil)
+  end
```

Comment:
@jsvd Yes thats right, but since we are using `#tap` and a block I dont see the need to take the full world `namespace` in that case `n` would be sufficient. :)


Let me try to get a proposal for that the block,  I am still wondering if we should allow something like this instead.

```ruby
agent.add_listerner(AgentMetricObserver.new(metric.namespace(:x))

#
emit(:pipeline_successfully_reloaded)
emit(:pipeline_reload_failures, e)
```